fulp_modules/features/antagonists/infiltrators/infil_objectives.dm

```diff
@@ -0,0 +1,204 @@
+
+//Animal Rights Consortium Objectives
+
+//pet killing
+
+/datum/objective/kill_pet
+	var/mob/living/target_pet
+	martyr_compatible = TRUE
+
+/datum/objective/kill_pet/proc/find_pet_target()
+	var/list/possible_target_pets = list (/mob/living/simple_animal/pet/dog/corgi/ian,
+	/mob/living/simple_animal/pet/fox/renault, /mob/living/simple_animal/pet/cat/runtime,
+	/mob/living/simple_animal/parrot/poly)
+
+	remove_duplicate(possible_target_pets) //removes pets from the list that are already in the owner's objective
+
+	var/chosen_pet = pick(possible_target_pets)
+	target_pet = locate(chosen_pet) in GLOB.mob_living_list
+
+	update_explanation_text()
+
+/datum/objective/kill_pet/proc/remove_duplicate(possible_target_pets)
+	for(var/datum/objective/kill_pet/objective in owner.get_all_objectives())
+		if(objective.target_pet.type in possible_target_pets)
+			possible_target_pets -= objective.target_pet.type
+
+
+/datum/objective/kill_pet/update_explanation_text()
+	..()
+	if(target_pet)
+		explanation_text = "[target_pet] has been tainted by Nanotrasen agenda, give them a mercy killing."
+
+
+/datum/objective/kill_pet/check_completion()
+	return completed || (target_pet.stat == DEAD)
+
+//scientist killing
+
+/datum/objective/assassinate/proc/find_sci_target()
+	var/list/sci_targets = list()
+	for(var/mob/living/carbon/human/player as anything in GLOB.human_list)
+		if(player.stat == DEAD)
+			continue
+		if((player.mind?.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_SCIENCE))
+			sci_targets += player.mind
+
+	if(!sci_targets.len)
+		find_target()
+		return
+	else
+		target = pick(sci_targets)
+
+	if(target?.current)
+		explanation_text = "Make a stance against science's animal experimentation by assassinating [target.name] the [!target_role_type ? target.assigned_role.title : target.special_role]!"
+
+//monkifying head of staff
+
+/datum/objective/assassinate/monkify
+
+/datum/objective/assassinate/monkify/proc/find_head_target()
+	var/list/com_targets = SSjob.get_all_heads()
+	if(!com_targets.len)
+		find_target()
+	else
+		target = pick(com_targets)
+	update_explanation_text()
+
+/datum/objective/assassinate/monkify/check_completion()
+	. = ..()
+
+	if(!.)
+		return FALSE
+
+	return completed || ismonkey(target.current)
+
+/datum/objective/assassinate/monkify/update_explanation_text()
+	..()
+	if(target?.current)
+		explanation_text = "Monkify [target.name] the [!target_role_type ? target.assigned_role.title : target.special_role] and euthanize him."
+
+
+//Mauradars Objectives
+
+//emagging communication console
+
+/datum/objective/emag_console
+	explanation_text = "Secure communication lines between Space Station 13 and the Syndicate by emagging the communications console!"
+
+/datum/objective/emag_console/check_completion()
+	var/check_emag = FALSE
+	for(var/obj/machinery/computer/communications/console in GLOB.machines)
+		if(!is_station_level(console.z))
+			continue
+		if(console.obj_flags & EMAGGED)
+			check_emag = TRUE
```

Comment:
this requires a console be emagged by the end of a round, it being deconstructed last second fucks it over. Would there have a better way to implement this? Bugging the console?

Reply:
replaced it with another objective